### ISSUE TITLE
[sw/silicon_creator] AST init check

### ISSUE THREAD

**Issue by alphan** [lowRISC/opentitan#9685]
In DEV/PROD* LC states, ROM or ROM_EXT SW should check that ast_init_done was indeed set. At those LC states, INIT_EN is expected to be true. I think that for the engineering samples, it is better to have this checking code in the ROM_EXT.

_Originally posted by @zi-v in https://github.com/lowRISC/opentitan/pull/9557#r767237389_

**Comment by alphan** [comment]
IIUC, there is no harm in continuing with `mask_rom_start.S` without checking `SENSOR_CTRL_STATUS_AST_INIT_DONE_BIT`. If that's the case, we can move that check from `mask_rom_start.S` to `mask_rom.c` and gate it based on the life cycle state (skip if TEST_UNLOCKED* or RMA).

@tjaychen, @moidx, @cfrantz, do we need to disable read access to the (memory mapped) AST init region in OTP?

**Comment by zi-v** [comment]
Need to disable read access to the (memory mapped) AST init region in OTP only while in DEV/PROD*

**Comment by tjaychen** [comment]
@zi-v are you okay with us pushing such disabling to `rom_ext`? I don't think that's something that HAS to be in rom right?

**Comment by zi-v** [comment]
Yes, under the assumption that only the chip creator can authorize/sign changes in the ROM_EXT

**Comment by tjaychen** [comment]
I believe that to be the case. @moidx @cfrantz @alphan please correct me if i am wrong.

**Comment by alphan** [comment]
> I believe that to be the case. @moidx @cfrantz @alphan please correct me if i am wrong.

Correct.

**Comment by tjaychen** [comment]
great! so can we make an official decision here to not HAVE to include this in ROM? :) 
Would you prefer that @alphan ?

**Comment by alphan** [comment]
> great! so can we make an official decision here to not HAVE to include this in ROM? :) Would you prefer that @alphan ?

Yes, created #9810 for locking `CREATOR_SW_CFG` (the partition that holds AST configuration) in ROM_EXT.

**Comment by alphan** [comment]
#9810 locks `CREATOR_SW_CFG` to disable read access to the AST init region.
#12068 moves the init done check to mask_rom.c
